src/Runtime.Base/src/System/Runtime/RuntimeExports.cs

```diff
@@ -216,7 +222,65 @@ public static unsafe void RhUnboxNullable(ref byte data, EETypePtr pUnboxToEETyp
             {
                 throw ptrUnboxToEEType->GetClasslibException(ExceptionIDs.InvalidCast);
             }
-            InternalCalls.RhUnbox(obj, ref data, ptrUnboxToEEType);
+            RhUnbox(obj, ref data, ptrUnboxToEEType);
+        }
+
+        [RuntimeExport("RhUnbox")]
+        public static unsafe void RhUnbox(object obj, ref byte data, EEType* pUnboxToEEType)
+        {
+            // When unboxing to a Nullable the input object may be null.
+            if (obj == null)
+            {
+                Debug.Assert(pUnboxToEEType != null && pUnboxToEEType->IsNullable);
+
+                // The first field of the Nullable is a Boolean which we must set to false in this case to indicate no
+                // value is present.
+                Unsafe.As<byte, bool>(ref data) = false;
```

Comment:
You can simplify this by zeroing out the whole thing. Zeroing out a few more bytes is going to be smaller and faster than doing the math with NullableType, NullableValueOffset, etc.

Reply:
Yeah, the code was trying to be too explicit. Done.